app/jest/my-dashboard/my-projects/roleManagement.test.tsx

```diff
@@ -67,120 +73,164 @@ describe('role management tab', () => {
     expect(screen.getByRole('tab', { selected: false, name: 'Test' }));
     expect(screen.getByRole('tab', { selected: false, name: 'Prod' }));
     await waitFor(() => {
-      expect(screen.getByRole('columnheader', { name: 'Role Name' })).toBeInTheDocument();
+      expect(screen.getByRole('row', { name: 'Role Name' })).toBeInTheDocument();
     });
     await waitFor(() => {
-      expect(screen.getByText('role1'));
+      expect(screen.findByRole('cell', { name: 'role1' }));
     });
     expect(screen.getByPlaceholderText('Search existing roles'));
   });
 
   it('Should be able to input keywords in Search Existing Role input field', async () => {
     render(<RoleManagement integration={{ ...sampleRequest }} />);
     const searchRoleInput = screen.findByRole('textbox');
-    fireEvent.change(await searchRoleInput, { target: { value: 'sample_role' } });
-    expect(screen.getByDisplayValue('sample_role')).toBeInTheDocument();
+    fireEvent.change(await searchRoleInput, { target: { value: 'role' } });
+    expect(screen.getByDisplayValue('role')).toBeInTheDocument();
+    fireEvent.click(screen.getByRole('button', { name: 'Search' }));
+    await waitFor(() => {
+      expect(screen.getByRole('cell', { name: 'role1' }));
+    });
+  });
+
+  it('Testing on the headers, buttons and input field in the modal', async () => {
+    render(<CreateRoleContent integrationId={sampleRequest.id as number} environments={['dev', 'test', 'prod']} />);
+    screen.getByRole('row', { name: 'Role Name Environments' });
+    fireEvent.click(screen.getByRole('img', { name: 'Add Role' }));
+    expect(screen.queryAllByTestId('role-name-input-field')).toHaveLength(2);
+    const removeRoleNameInput = screen.getAllByRole('img', { name: 'Remove Role' });
+    fireEvent.click(removeRoleNameInput[1]);
+    expect(screen.queryAllByTestId('role-name-input-field')).toHaveLength(1);
   });
 
   it('Should be able to input keywords in Create New Role input field', async () => {
-    render(<RoleManagement integration={{ ...sampleRequest }} />);
+    render(<CreateRoleContent integrationId={1} environments={['dev']} />);
+
```

Comment:
@kurokn are you validating `save` button callout when creating a new role??

Reply:
@kurokn the `save` is making a callout to `bulkCreateRole` function at `./sso-requests/app/page-partials/my-dashboard/RoleManagement/CreateRoleContent.tsx` through use of a `ref` and `useImperativeHandle` hooks. So try to mock that function and let me know if that works?